compiler/rustc_expand/src/mbe/quoted.rs

```diff
@@ -197,7 +237,19 @@ fn parse_tree(
                     }
                 }
 
-                // `tree` is followed by a random token. This is an error.
+                // `tree` is followed by another `$`. This is an escaped `$`.
+                Some(tokenstream::TokenTree::Token(Token { kind: token::Dollar, span })) => {
+                    maybe_emit_macro_metavar_expr_feature(features, sess, span);
```

Comment:
It only makes sense to emit the feature error in the `if !parsing_patterns` case, otherwise it's mostly a useless noise.

Reply:
Hehehe... :P

Sorry, I will fix it in a following PR

Reply:
`if !parsing_patterns` triggered several errors on `dollar-dollar-has-correct-behavior.rs` and surpressed the error that would be emitted in

```rust
macro_rules! dollar_dollar_in_the_lhs {
    ( $$ $a:ident ) => {
    };
}
```